internal/metrics/metrics.go

```diff
@@ -230,6 +241,15 @@ func (m *Metrics) Zero() {
 	prometheus.NewTimer(m.CacheHandlerOnUpdateSummary).ObserveDuration()
 }
 
+// SetBuildInfo sets the Contour Build Info
+func (m *Metrics) SetBuildInfo(version string) {
```

Comment:
Not particularly proud of this. If a more eloquent solution works for this I am open to it.

Reply:
You don't need this at all. Just create the gauge in `NewMetrics`, and remove this.